firmware/thruna/BMS/Src/App/states/App_PreChargeState.c

```diff
@@ -85,17 +81,13 @@ static void PreChargeStateRunOnTick100Hz(struct StateMachine *const state_machin
             App_SharedStateMachine_SetNextState(state_machine, next_state);
         }
 
-        if (external_shutdown_occurred)
+        if (external_shutdown_occurred && is_charger_connected)
         {
+            // TODO: Why do we need this transition?
```

Comment:
I'm not 100% on this, but I think this allows for an external shutdown condition to transition to fault state BEFORE enabling the charger as part of ChargeStateRunOnEntry. This could be re-worked, but maybe save that for a separate PR.

Reply:
Ok sounds good, let's leave it for a follow up.